### PR TITLE
Communicate with location(s)

### LINKED ISSUE #8: Reliable UDP Template
Easy to use class the provides reliable UDP communication

### PR DESCRIPTION
**Pull request by prince-chrismc**
# Description
new method `sendTo` for communication with one or more locations

Fixes #8
- New method

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [ ] Examples
- [x] Test App

**Test Configuration**:
* OS: win7
* Compiler: javac
* Version: 8 something

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: DEMS-Common/src/Models/Location.java [hunk 1]
```diff
@@ -31,65 +31,81 @@ of this software and associated documentation files (the "Software"), to deal
...
-    	
-    	for (Location region : Location.values()) {
-    		if ((region.getPrefix() == null ? location == null : region.getPrefix().equalsIgnoreCase(location))
-    				|| (region.toString() == null ? location == null : region.toString().equalsIgnoreCase(location))) {
-    			return region;
-    		}
-    	}
-    	
-    	throw new Exception("Invalid Location!");
+    public static Location fromString(String location) {
```
Comment (Winterhart) [inline DEMS-Common/src/Models/Location.java]: Nice config `enum`

File: DEMS-Common/src/UDP/Message.java [hunk 2]
```diff
@@ -55,6 +57,7 @@ public Message(OperationCode code, int seq, String data, AddressBook addrInfo) t
         
         this.m_Code = code;
         this.m_SeqNum = seq;
+        this.m_Location = Location.INVALID;
```
Comment (Winterhart) [inline DEMS-Common/src/UDP/Message.java]: The default location is `INVALID` ok

File: DEMS-Common/src/UDP/Socket.java [hunk 3]
```diff
@@ -31,17 +32,17 @@
 /**
  *
  * @author cmcarthur
- * 
+ *
```
Comment (Winterhart) [inline DEMS-Common/src/UDP/Socket.java]: Would probably use this example...today thanks

File: DEMS-Common/src/Models/Location.java (hunk 1 above)
Comment (prince-chrismc) [reply DEMS-Common/src/Models/Location.java]: easiest way of working =p

**Pull request merged**